### PR TITLE
feat(pci.project.billing): open monthly confirmation in a modal

### PR DESCRIPTION
**Pull request by marie-j**
Solves MANAGER-2571

### CODE REVIEW

File: packages/manager/modules/pci/src/components/project/billing/hourly-resource-list/controller.js [hunk 1]
```diff
@@ -1,20 +1,23 @@
 
-export default /* @ngInject */ function (
-  DetailsPopoverService,
-) {
-  const self = this;
+export default /* @ngInject */ class {
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/components/project/billing/hourly-resource-list/controller.js]: Extra empty line.

File: packages/manager/modules/pci/src/components/project/billing/hourly-resource-list/controller.js [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
+  /* @ngInject */
+  constructor(DetailsPopoverService) {
```
Comment (frenautvh) [inline packages/manager/modules/pci/src/components/project/billing/hourly-resource-list/controller.js]: i'm not used to see uppercased variable names outside of class declaration, did i miss something?

File: packages/manager/modules/pci/src/projects/project/billing/billing.controller.js [hunk 3]
```diff
@@ -50,4 +50,11 @@ export default class CloudProjectBillingConsumptionCurrentCtrl {
   refreshMessages() {
     this.messages = this.messageHandler.getMessages();
   }
+
+  switchToMonthly({ instanceId }) {
+    this.$state.go('pci.projects.project.billing.monthly', {
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/billing/billing.controller.js]:
```suggestion
    return this.$state.go('pci.projects.project.billing.monthly', {
```

File: packages/manager/modules/pci/src/components/project/billing/instance-list/template.html [hunk 4]
```diff
@@ -48,48 +47,14 @@
...
-                                            write-right-required>
-                                        {{ :: 'cpbc_hourly_instance_pass_to_monthly' | translate }}
-                                    </button>
-                                </div>
+                                <button type="button"
+                                        data-ng-if="$ctrl.switchToMonthlyBilling && instanceConsumption.monthlyBilling === null && !instanceConsumption.isDeleted"
+                                        class="oui-button oui-button_secondary"
+                                        data-ng-click="$ctrl.switchToMonthlyBilling({ instance: instanceConsumption })"
+                                        write-right-required>
+                                    {{ :: 'cpbc_hourly_instance_pass_to_monthly' | translate }}
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/components/project/billing/instance-list/template.html]: `span`

File: packages/manager/modules/pci/src/components/project/billing/instance-list/template.html [hunk 5]
```diff
@@ -48,48 +47,14 @@
...
-                                            data-ng-disabled="$ctrl.loaders.monthlyBilling"
-                                            write-right-required>
-                                        {{ :: 'cpbc_hourly_instance_pass_to_monthly' | translate }}
-                                    </button>
-                                </div>
+                                <button type="button"
+                                        data-ng-if="$ctrl.switchToMonthlyBilling && instanceConsumption.monthlyBilling === null && !instanceConsumption.isDeleted"
+                                        class="oui-button oui-button_secondary"
+                                        data-ng-click="$ctrl.switchToMonthlyBilling({ instance: instanceConsumption })"
+                                        write-right-required>
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/components/project/billing/instance-list/template.html]:
```suggestion
                                        data-write-right-required>
```

File: packages/manager/modules/pci/src/projects/project/billing/billing.html [hunk 6]
```diff
@@ -59,7 +59,10 @@ <h2 data-translate="cpbc_tab_consumption"></h2>
                         <span data-translate="cpbc_hourly_header_description"></span>
                         &nbsp;(<span data-ng-bind="BillingCtrl.data.totals.hourly.total.toFixed(2) + ' ' + BillingCtrl.data.totals.currencySymbol"></span>)
                     </p>
-                    <hourly-resource-list resources="BillingCtrl.data" show-switch-to-monthly-billing-option="true"></hourly-resource-list>
+                    <hourly-resource-list
+                        resources="BillingCtrl.data"
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/billing/billing.html]: `data-` and right above too :)

**Pull request merged**